compiler/noirc_evaluator/src/ssa/function_builder/data_bus.rs

```diff
@@ -142,7 +142,12 @@ impl FunctionBuilder {
         assert!(databus.databus.is_none(), "initializing finalized call data");
         let typ = self.current_function.dfg[value].get_type().into_owned();
         match typ {
-            Type::Numeric(_) => {
+            Type::Numeric(numeric_type) => {
+                let value = if matches!(numeric_type, NumericType::NativeField) {
+                    value
+                } else {
+                    self.insert_cast(value, NumericType::NativeField)
```

Comment:
This is going to break if we have signed integers in the data bus after https://github.com/noir-lang/noir/pull/8818.

Not sure if we already ban that or if that will cause downstream issues cc @guipublic

Reply:
Oooh... good point. I checked the current `insert_safe_cast` and it's fine casting anything to Field, but I forgot about #8818.

I wonder if that implies that signed integers can't be in the databus.